mmv1/products/workflows/api.yaml

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Workflows
+display_name: Workflows
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://workflows.googleapis.com/v1beta/
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://workflows.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Workflows API
+    url: https://console.cloud.google.com/apis/library/workflows.googleapis.com/
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - True
+      - False
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Workflow'
+    description: |
+      Workflow program to be executed by Workflows.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Managing Workflows':
+          'https://cloud.google.com/workflows/docs/creating-updating-workflow'
+      api: 'https://cloud.google.com/workflows/docs/reference/rest/v1/projects.locations.workflows'
+    base_url: projects/{{project}}/locations/{{region}}/workflows
+    create_url: projects/{{project}}/locations/{{region}}/workflows?workflowId={{name}}
+    self_link: '{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: region
+        description: The region of the workflow.
+        url_param_only: true
+        input: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: Name of the Workflow.
```

Comment:
should this also be         url_param_only: true
?

Reply:
The name returned in the api response is a the resource reference (fully quantified) I suspect you'll get a continuous diff using this. Could you maybe add an import test?

Reply:
I tried import tests but it doesn't work. I set the import_format to `projects/{{project}}/locations/{{region}}/workflows/{{name}}`

Importing doesn't make sense as most of the user settable fields are immutable anyway.

I can tweak it to use workflow_id but I don't like that as you end up with 3 pseudo indentical `workflow_id`, `name` and `id` fields and the last 2 being identical.

Reply:
Can you validate locally that you aren't getting a continuous diff after apply? pending that then I am okay with this

Reply:
I did and there is not diff. There is a new multistep test that I just pushed.